File: wellness/wellness.Service/Database/Reservation.cs
using System;
using System.Collections.Generic;

namespace wellness.Service.Database;

public partial class Reservation
{
    public int Id { get; set; }

    public int UserId { get; set; }

    public string Date { get; set; } = null!;

    public string Time { get; set; } = null!;

    public bool Status { get; set; }

    public int TreatmentId { get; set; }

    public virtual Treatment Treatment { get; set; } = null!;

    public virtual User User { get; set; } = null!;
}

File: wellness/UI/mobile/lib/models/reservation.dart
import 'package:json_annotation/json_annotation.dart';

part 'reservation.g.dart';

@JsonSerializable()
class Reservation {
  final int id;
  final String firstName;
  final String lastName;
  final String date;
  final String time;
  final bool? status;
  final String treatment;
  final int treatmentId;

  Reservation(this.id, this.firstName, this.lastName, this.date, this.time,
      this.status, this.treatment, this.treatmentId);
  factory Reservation.fromJson(Map<String, dynamic> json) =>
      _$ReservationFromJson(json);

  Map<String, dynamic> toJson() => _$ReservationToJson(this);
}
